Source/WebCore/rendering/updating/RenderTreeUpdater.cpp

```diff
@@ -327,6 +327,8 @@ void RenderTreeUpdater::updateElementRenderer(Element& element, const Style::Ele
         element.setDisplayContentsChanged(false);
     }
 
+    element.updateContentVisibility(elementUpdateStyle.contentVisibility());
```

Comment:
RenderTreeUpdater putting style data into DOM like this is pretty nasty. I think this feature is best implemented as a fake inherited property in RenderStyle, similar to effectiveInert.